### PR TITLE
fix(curriculum): grammar issues and updated test in step 29 of music player project

### LINKED ISSUE #53700: fix grammar issues and update tests in step 29 of music player project
### Describe the Issue

There are a few grammar issues in this description. Plus the description can be broken up with spacing a little bit better for readability. 

## updated description

```md
Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property on the `audio` object. 

Add an `if` statement to check whether the `userData?.currentSong` is falsy *OR* if `userData?.currentSong.id` is strictly not equal to `song.id`. This condition will check if no current song is playing or if the current song is different from the one that is about to be played.

Inside your `if` block, set the `audio.currentTime` property to `0`.
```
Also the tests are looking for a specific answer of `userData?.currentSong === null || userData?.currentSong.id !== song.id` for the condition.
But I don't see why we need to specifically check for `userData?.currentSong === null` when either of these options would be valid

```js
!userData?.currentSong || userData?.currentSong.id !== song.id

userData?.currentSong === null || userData?.currentSong.id !== song.id

userData?.currentSong === undefined || userData?.currentSong.id !== song.id
```

The tests should be more flexible to allow any valid answer.

Also, order shouldn't matter here because we want to test that campers understand the concept. Not forcing a particular order here.

This should pass but currently doesn't
```js
if(userData?.currentSong.id !== song.id||userData?.currentSong === null)
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-string-and-array-methods-by-building-a-music-player/step-29

### Your code

see explanation above

### Expected behavior

see explanation above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by RGHANILOO**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Hi @RGHANILOO ! 

It looks like there are a couple of issues with this PR.

The first issue is that your current PR doesn't have any files changed
Screenshot 2024-02-28 at 4 03 54 PM

Please push up another commit to add the changes you want to make so a review can be done.

The second issue is the PR title. It isn't clear what these changes are about.
It is best to use descriptive PR titles so matinainers know what the PR's are about.

A good title would be the following:
```md
fix(curriculum): grammar issues and update tests in step 29 of music player project
```
once those changes are applied, then we can properly review your PR 👍 
If you need help with these changes, then please reach out on the contributors channel on discord or the contributors subforum. 👍

**Comment by RGHANILOO** [comment]
Hi 

Thanks @jdwilkin4 for your feedback.
not sure what happened the first time round, i have now made the necessary changes again as highlighted.
Thanks

**Comment by jdwilkin4** [comment]
@RGHANILOO 

It looks like some the tests are failing.
Can you please look into that?
It looks like the issue has to deal with the changes you made to one of the tests for the music player

**Comment by RGHANILOO** [comment]
> @RGHANILOO
> 
> It looks like some the tests are failing. Can you please look into that? It looks like the issue has to deal with the changes you made to one of the tests for the music player

Hey @jdwilkin4 ,
i have made the appropriate changes based on a suggestion on another PR on this issue  that satisfies all the described conditions.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md [hunk 1]
```diff
@@ -7,17 +7,19 @@ dashedName: step-32
...
 
 
 # --hints--
 
-You should create an `if` statement with the condition `userData?.currentSong === null || userData?.currentSong.id !== song.id`.
+You should create an `if` statement checking `userData?.currentSong` is either `null` or `undefined`, or if its `id` is not equal to `song.id`.
 
 ```js
-assert.match(code, /if\s*\(\s*userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*\)\s*\{\s*/)
+assert.match(code, /if\s*\(\s*(?:userData\?\.currentSong\s*===\s*(?:null|undefined)\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id|userData\?\.currentSong\.id\s*!==\s*song\.id\s*\|\|\s*userData\?\.currentSong\s*===\s*(?:null|undefined))\s*\)\s*\{\s*/);
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md:22]:
```suggestion
assert.match(code, /if\s*\(\s*(?:userData\?\.currentSong\s*===\s*(?:null|undefined)\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id|userData\?\.currentSong\.id\s*!==\s*song\.id\s*\|\|\s*userData\?\.currentSong\s*===\s*(?:null|undefined))\s*\)\s*\{/);
```
Nitpick

**Review by gikf — COMMENTED** [review]
Second test also needs to be changed, to add more lenient regex to it as well.

**Comment by gikf** [comment]
@RGHANILOO, sorry, I should be more clear. The second test is this one:
https://github.com/freeCodeCamp/freeCodeCamp/blob/c628dbca7dd0d3a41c1331b5741a6760b23c634e/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md?plain=1#L25-L29
Without updating it with similar pattern as first one, it will be able for condition that passes first test not pass second.

**Comment by jdwilkin4** [comment]
It looks like the steps have been updated and the changes should now be updated to step 38. not step 32.
I am close this PR.

I would suggest you do following:
- delete this branch
- update your main branch with the latest changes
- create a new branch
- create a new fresh PR with these changes and we will be happy to review that 👍

**Pull request closed without merging**